src/domains/chat/services/prompt.service.ts

```
@@ -409,6 +495,150 @@ Optimized search query:
+      throw new ResourceExhaustedError(
+        'input size',
+        MAX_INPUT_SIZE,
+        input.length,
+      );
+    }
+
+    // Check for null bytes and control characters
+    if (
+      input.includes('\0') ||
+      // eslint-disable-next-line no-control-regex
+      /[\x00-\x08\x0B\x0C\x0E-\x1F\x7F]/.test(input)
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)